internal/olm/operator/registry/operator_installer.go

```diff
@@ -208,15 +209,17 @@ func (o OperatorInstaller) getOperatorGroup(ctx context.Context) (*v1.OperatorGr
 	return &ogList.Items[0], true, nil
 }
 
-func (o OperatorInstaller) createSubscription(ctx context.Context, cs *v1alpha1.CatalogSource) (*v1alpha1.Subscription, error) {
+func (o OperatorInstaller) createSubscription(ctx context.Context, csName string) (*v1alpha1.Subscription, error) {
 	sub := newSubscription(o.StartingCSV, o.cfg.Namespace,
 		withPackageChannel(o.PackageName, o.Channel, o.StartingCSV),
-		withCatalogSource(cs.GetName(), o.cfg.Namespace),
+		withCatalogSource(csName, o.cfg.Namespace),
 		withInstallPlanApproval(v1alpha1.ApprovalManual))
```

Comment:
Exactly what I described above regarding Options pattern. So now you can clearly see that we are creating a `Subscription` instance and setting it's `PackageName`, `CatalogSource`, and `InstallPlan`. And if we want to update a fourth attribute, we can just add a new `withThatNewAttribute` and not have to change the `newSubscription`.

Reply:
This makes SO much more sense now.